README.md

```diff
@@ -3,4 +3,281 @@
+
+```json
+{
+  "orgName": "SomeOrgName",
+  "edition": "Developer",
+  "features": ["EnableSetPasswordInApi", "Functions"],
+  "settings": {
+    "lightningExperienceSettings": {
+      "enableS1DesktopEnabled": true
+    },
+    "mobileSettings": {
+      "enableS1EncryptedStoragePref2": false
+    }
+  }
+}
+```
+
+The above is needed for connecting and deploying Functions in scratch orgs which is 
+[the recommended workflow](https://developer.salesforce.com/docs/platform/functions/guide/connect-dev-org.html) when working 
+with Functions.
+
+And, to deploy any Salesforce Function, your SFDX project needs to be a git repo.  This is because the deployment process 
+uses git tracked changes to figure out what to deploy.  Run the following commands to setup git:
+
+```sh
+git init
+```
+
+> It is not a requirement to push your code to GitHub or any other code hosting site.  Committing locally will work fine for deploying.  
+
+### Connect Your Org
+
+You’ll need to configure your Salesforce org to develop and invoke Salesforce Functions. Develop your functions in scratch 
+orgs with Dev Hub or in sandbox orgs.  Follow the steps on 
+[this page](https://developer.salesforce.com/docs/platform/functions/guide/configure_your_org.html) to ensure everything 
+is setup correctly first.
+
+Once your Org is configured you can login and set it as the default Dev Hub with the following command:
+
+```sh
+sf login org --alias PythonOrg --set-default-dev-hub --set-default
+```
+
+This will make **PythonOrg** the default Dev Hub for subsequent commands.
+
+Then create a scratch org:
+
+```sh
+sfdx force:org:create \
+  --definitionfile config/project-scratch-def.json \
+  --setalias PythonScratch \
+  --setdefaultusername
+```
+
+This will make it so that when you run a Function, it will connect to and use the **PythonScratch** org.
+
+### Connect Your Compute Environment
+
+Login to Salesforce Functions with the same credentials you used to connect your Dev Hub org.
+
+```sh
+sf login functions
+```
+
+Then you will be able to create the compute environment and associate it with the **PythonScratch** org we created while
+connecting your org.
+
+```sh
+sf env create compute \
+  --connected-org PythonScratch \
+  --alias PythonCompute
+```
+
+This will make it so that when you deploy a Function, it will be deployed to the **PythonCompute** environment linked to 
+your scratch org.
+
+## Create and Run a Python Function Locally
+
+### Generate the Python Function
+
+From the SFDX project root, run:
+
+```sh
+sf generate function \
+  --language python \
+  --function-name hellopython
+```
+
+The remaining commands will be executed within the newly generated project folder change to that folder by running
+
+```sh
+cd functions/hellopython
+```
+
+### Create the Python [Virtual Environment](https://docs.python.org/3/library/venv.html#creating-virtual-environments) & Install Dependencies
+
+```sh
+python3 -m venv venv
+source venv/bin/activate
```

Comment:
On Windows unfortunately the command/path for activating the virtual environment is slightly different.

I just remembered we have docs for creating/activating venvs in the Heroku Python getting started guide, if it's helpful for inspiration?
https://devcenter.heroku.com/articles/getting-started-with-python?singlepage=true#install-app-dependencies-locally